source/multigrid/mg_transfer_internal.cc

```diff
@@ -191,25 +191,47 @@ namespace internal
             tria->level_ghost_owners();
           std::map<int, std::vector<DoFPair>> send_data;
 
-          // * find owners of the level dofs and insert into send_data
-          // accordingly
-          for (const auto &dofpair : send_data_temp)
+          for (unsigned int level = 0; level < n_levels; ++level)
             {
-              std::set<types::subdomain_id>::iterator it;
-              for (it = neighbors.begin(); it != neighbors.end(); ++it)
-                {
-                  if (mg_dof
-                        .locally_owned_mg_dofs_per_processor(dofpair.level)[*it]
-                        .is_element(dofpair.level_dof_index))
-                    {
-                      send_data[*it].push_back(dofpair);
-                      break;
-                    }
-                }
-              // Is this level dof not owned by any of our neighbors? That would
-              // certainly be a bug!
-              Assert(it != neighbors.end(),
-                     ExcMessage("could not find DoF owner."));
+              const IndexSet &is_local = mg_dof.locally_owned_mg_dofs(level);
+
+              std::vector<unsigned int> level_dof_indices;
+              std::vector<unsigned int> global_dof_indices;
+              for (const auto &dofpair : send_data_temp)
+                if (dofpair.level == level)
+                  {
+                    level_dof_indices.push_back(dofpair.level_dof_index);
+                    global_dof_indices.push_back(dofpair.global_dof_index);
+                  }
+
+              std::vector<std::pair<unsigned int, unsigned int>> order;
+              {
+                int c = 0;
+                for (auto i : level_dof_indices)
+                  {
+                    order.push_back({i, c++});
```

Comment:
```
clang-tidy-6.0 [...] source/multigrid/mg_transfer_internal.cc:212:27: 
error: use emplace_back instead of push_back [modernize-use-emplace,-warnings-as-errors]
```

Reply:
@masterleinad: thanks for the hint!